dev/lint-commits.sh

```diff
@@ -19,14 +19,21 @@ fi
 BASE_COMMIT="$1"
 HEAD_COMMIT="$2"
 
-# git diff --check
-# uses .gitattributes to know what to check
-if git diff --check "$BASE_COMMIT" "$HEAD_COMMIT";
+bad=()
+while IFS= read -r commit; do
+    echo Checking "$commit"
+    # git diff --check
+    # uses .gitattributes to know what to check
+    if ! git diff --check "${commit}^" "$commit";
+    then
+        bad+=("$commit")
+    fi
+done < <(git rev-list "$HEAD_COMMIT" --not "$BASE_COMMIT" --)
```

Comment:
Previously used rev-list was incorrect, it listed all commits accessible from either the PR or the merge base (ie the vast majority of commits).